pkg/health/server/prober.go

```diff
@@ -216,11 +214,31 @@ func (p *prober) setNodes(added nodeMap, removed nodeMap) {
 					IP: elem.IP,
 				}
 			}
-			p.results[ip].Icmp = result
 		}
 	}
 }
 
+func (p *prober) updateIcmpStatus() {
+	p.Lock()
+	defer p.Unlock()
+
+	for _, status := range p.results {
+		if status.Icmp == nil {
+			status.Icmp = &models.ConnectivityStatus{}
+			status.Icmp.Status = "Connection timed out"
+		}
+	}
+}
+
+func (p *prober) cleanIcmpStatus() {
```

Comment:
Nit: "Clear" is more accurate here, but this is a very nitty nit.

Reply:
done and done